### DISCUSSION TITLE
I'm having trouble after the last update, 2.5.0 --> 2.5.2

### DISCUSSION THREAD

**Discussion by gitkevin** [abraunegg/onedrive#2985] [Q&A] [answered]
* I was required to update my Linux password, and did so.
  * My changed password required me to run 'onedrive --reauth', which I did.
  * "Authorise this application by visiting:
     https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id ... etc.
     Which I did.
  * The above step created a very long URI in the URL space of my browser.
    https://login.microsoftonline.com/common/oauth2/nativeclient?code=1.AQgA ... etc.
  * I copied the full URI from my browser and pasted it at the input prompt, below.
    Enter the response uri from your browser:
  * After entering the URI I got the following responses.
    The application has been successfully authorised, but no extra command options have been specified.
    Please use 'onedrive --help' for further assistance in regards to running this application.

  * systemctl start onedrive
    ==== AUTHENTICATING FOR org.freedesktop.systemd1.manage-units ====
    Authentication is required to start 'onedrive.service'.
    Authenticating as: My Name, My Role (myusername)
    Password:

    ==== AUTHENTICATION COMPLETE ====
    myusername@myhost:~$ systemctl status onedrive
      onedrive.service - OneDrive Client for Linux
         Loaded: loaded (/usr/lib/systemd/system/onedrive.service; enabled; preset: disabled)
         Active: active (running) since Thu 2024-11-14 17:34:58 UTC; 887ms ago
           Docs: https://github.com/abraunegg/onedrive
       Main PID: 848083 (onedrive)
          Tasks: 2 (limit: 22972)
         Memory: 3.5M
            CPU: 181ms
         CGroup: /system.slice/onedrive.service
                 └─848083 /usr/bin/onedrive --monitor

    Nov 14 17:34:58 myhost.mydomain systemd[1]: Started OneDrive Client for Linux.
    Nov 14 17:34:58 myhost.mydomain onedrive[848083]: Using IPv4 and IPv6 (if configured) for all network operations
    Nov 14 17:34:58 myhost.mydomain onedrive[848083]: Attempting to contact Microsoft OneDrive Login Service
    Nov 14 17:34:59 myhost.mydomain onedrive[848083]: Successfully reached Microsoft OneDrive Login Service

**AT THIS POINT ALL LOOKS GOOD.**

  * myusername@myhost:~$ **systemctl status onedrive**
  onedrive.service - OneDrive Client for Linux
     Loaded: loaded (/usr/lib/systemd/system/onedrive.service; enabled; preset: disabled)
     Active: active (running) since Thu 2024-11-14 17:34:58 UTC; 887ms ago
       Docs: https://github.com/abraunegg/onedrive
   Main PID: 848083 (onedrive)
      Tasks: 2 (limit: 22972)
     Memory: 3.5M
        CPU: 181ms
     CGroup: /system.slice/onedrive.service
             └─848083 /usr/bin/onedrive –monitor

  * Nov 14 17:42:12 myhost.mydomain onedrive[848730]: **The OneDrive API could not be initialised**
    Nov 14 17:42:13 myhost.mydomain systemd[1]: onedrive.service: Main process exited, code=exited, status=1/FAILURE
    Nov 14 17:42:13 myhost.mydomain systemd[1]: onedrive.service: Failed with result 'exit-code'.
    Nov 14 17:42:16 myhost.mydomain systemd[1]: onedrive.service: Scheduled restart job, restart counter is at 430.
    Nov 14 17:42:16 myhost.mydomain systemd[1]: Stopped OneDrive Client for Linux.
    Nov 14 17:42:16 myhost.mydomain systemd[1]: Started OneDrive Client for Linux.
    Nov 14 17:42:16 myhost.mydomain onedrive[848737]: Using IPv4 and IPv6 (if configured) for all network operations
    Nov 14 17:42:16 myhost.mydomain onedrive[848737]: Attempting to contact Microsoft OneDrive Login Service
    Nov 14 17:42:16 myhost.mydomain onedrive[848737]: Successfully reached Microsoft OneDrive Login Service
    Nov 14 17:42:17 myhost.mydomain onedrive[848737]: **WARNING: Your onedrive client version is now obsolete and unsupported.
                                                            Please upgrade your client version.**
    Nov 14 17:42:17 myhost.mydomain onedrive[848737]: Current Application Version: 2.5.0
    Nov 14 17:42:17 myhost.mydomain onedrive[848737]: Version Available:           2.5.2
    Nov 14 17:42:17 myhost.mydomain onedrive[848737]: Configuring Global Azure AD Endpoints
    Nov 14 17:42:17 myhost.mydomain onedrive[848737]: Authorise this application by visiting:
    Nov 14 17:42:17 myhost.mydomain onedrive[848737]: https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id ... etc.
    Nov 14 17:42:17 myhost.mydomain onedrive[848737]: **Enter the response uri from your browser: An empty or invalid response uri was entered**
    Nov 14 17:42:17 myhost.mydomain onedrive[848737]: The OneDrive API could not be initialised
    Nov 14 17:42:18 myhost.mydomain systemd[1]: onedrive.service: Main process exited, code=exited, status=1/FAILURE
    Nov 14 17:42:18 myhost.mydomain systemd[1]: onedrive.service: Failed with result 'exit-code'.

  * I suspect the following to be of major importance
    **WARNING: Your onedrive client version is now obsolete and unsupported.
          Please upgrade your client version.**
    Current Application Version: 2.5.0
    Version Available:           2.5.2

  * With OneDrive version 2.5.0 I used onedrive-2.5.0-1.el9 on Rocky 9,
    and that worked great for months.
  * OneDrive told me to update to 2.5.2, and I attempted that as 2.5.0
    wasn't working for me anymore. I updated as follows:
    * sudo dnf remove onedrive
    * sudo yum localinstall onedrive-2.5.2-1.el9.x86_64.rpm
    * Authorise this application by visiting:
    * https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id ... etc.
    * Enter the response uri from your browser: **An empty or invalid response uri was entered
    * The OneDrive API could not be initialised**
    * onedrive.service: Main process exited, code=exited, status=1/FAILURE
    * onedrive.service: Failed with result 'exit-code'.

  * I see that **the current version changed to 2.5.3, and I can either
    build OneDrive myself, or wait until there's an 2.5.3 el9 package.**
  * The 2.5.2 version is listed on the [install.md](https://github.com/abraunegg/onedrive/blob/master/docs/install.md) page. I'll check back from time to time and see if a Fedora 2.5.3 version shows up.

**Comment by abraunegg** [comment]
@gitkevin 
Your post is a bit hard to follow ....

Essentially - when you change your Microsoft password you need to issue a `--reauth` to re-authenticate the client which is explained here: https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#handling-a-microsoft-onedrive-account-password-change

When upgrading the client on Rocky Linux, there is no need to remove the existing installation - all you need to do is issue a `dnf update` command. In saying this, your process simply should have been:

* `dnf -y update`
* `onedrive --reauth`
* `systemctl --user restart onedrive`

You have some other issue in the background:
```
Nov 14 17:42:13 myhost.mydomain systemd[1]: onedrive.service: Main process exited, code=exited, status=1/FAILURE
Nov 14 17:42:13 myhost.mydomain systemd[1]: onedrive.service: Failed with result 'exit-code'.
Nov 14 17:42:16 myhost.mydomain systemd[1]: onedrive.service: Scheduled restart job, restart counter is at 430.
```
This points to an erroneous systemd service running .. please ensure you have all the correct systemd services loaded and the ones not needed disabled / deleted / removed.

> wait until there's an 2.5.3 el9 package

These have been built - you can always check the Koji build system: https://koji.fedoraproject.org/koji/packageinfo?packageID=26044

![image](https://github.com/user-attachments/assets/c866bbe2-c24d-4aac-a8df-aaaf0f0b058b)

**Reply by gitkevin** [reply, marked as answer]
> @gitkevin Your post is a bit hard to follow ....

Sorry 'bout that.

> Essentially - when you change your Microsoft password you need to issue a `--reauth` to re-authenticate the client which is explained here: https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#handling-a-microsoft-onedrive-account-password-change

I saved the link.

> When upgrading the client on Rocky Linux, there is no need to remove the existing installation - all you need to do is issue a `dnf update` command. In saying this, your process simply should have been:
> 
> * `dnf -y update`
> * `onedrive --reauth`
> * `systemctl --user restart onedrive`

I've been using `onedrive --reauth`.  This works for me: `systemctl restart onedrive@myusername`.

> You have some other issue in the background:
> ```
> Nov 14 17:42:13 myhost.mydomain systemd[1]: onedrive.service: Main process exited, code=exited, status=1/FAILURE
> Nov 14 17:42:13 myhost.mydomain systemd[1]: onedrive.service: Failed with result 'exit-code'.
> Nov 14 17:42:16 myhost.mydomain systemd[1]: onedrive.service: Scheduled restart job, restart counter is at 430.
> ```
> This points to an erroneous systemd service running .. please ensure you have all the correct systemd services loaded and the ones not needed disabled / deleted / removed.

The system-wide onedrive was running in addition to onedrive@myusername. I shut down the system-wide instance, and onedrive@myusername is working quite well now.

> > wait until there's an 2.5.3 el9 package

2.5.3 el9 appeared in the Koji table while I was AFK.  Nice timing!  :-) 

> These have been built - you can always check the Koji build system: https://koji.fedoraproject.org/koji/packageinfo?packageID=26044
> 
> ![image](https://private-user-images.githubusercontent.com/4956234/387817574-c866bbe2-c24d-4aac-a8df-aaaf0f0b058b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwNTYxNzQsIm5iZiI6MTczMjA1NTg3NCwicGF0aCI6Ii80OTU2MjM0LzM4NzgxNzU3NC1jODY2YmJlMi1jMjRkLTRhYWMtYThkZi1hYWFmMGYwYjA1OGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTlUMjIzNzU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTE5Yjg0YWYxZmZlYzBhMWQ3MjFlZjVlZDgxZjVjM2Y3Njc3ODdlMGIwM2I0MzE5Mzc5ZjAxMmY5MWI5MDc1NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.rtp4sA76FVqwV7tAdPseAipXLBWsEdJ-0Tx46h2hJLU)

I tested to make sure that files appear and disappear the right way, and do so correctly from the server side and from the client side. It's looking good!

Thanks @abraunegg for the coaching!
